networkx/classes/tests/test_reportviews.py

```diff
@@ -596,8 +596,18 @@ def test_getitem(self):
         assert ev[0, 1] == {"foo": "bar"}
 
         # slicing
-        with pytest.raises(nx.NetworkXError):
+        with pytest.raises(nx.NetworkXError) as e:
             G.edges[0:5]
+        assert (
+            str(e.value)
+            == f"{type(G.edges).__name__} does not support slicing, try list(G.edges)[0:5:None]"
+        )
+
+        # Invalid edge
+        with pytest.raises(KeyError) as e:
+            G.edges[0, 9]
+
+        assert e.value.args[0] == "The edge (0, 9) is not in the graph."
```

Comment:
There is an argument to pytest.raises that allows you to test matching the error message:
```suggestion
        with pytest.raises(KeyError, match="The edge (0, 9) is not in the graph."):
            G.edges[0, 9]
```
The same can be done to the first test.  And the match argument doesn't need to exactly match. It can be a substring, and it also accepts/uses regular expressions. See the pytest.raises docs if you are interested.

Reply:
Very nice! Didn't know about that one.